### PR TITLE
REFACTOR-#6815: move experimental parsers into 'modin.experimental' folder

### LINKED ISSUE #6815: Move experimental parsers into 'modin.experimental' folder
To make it more consistent, given that the dispatchers are located in the experimental folder.

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6815 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/core/execution/dask/implementations/pandas_on_dask/io/io.py [hunk 1]
```diff
@@ -49,6 +46,11 @@
     ExperimentalPickleDispatcher,
     ExperimentalSQLDispatcher,
 )
+from modin.experimental.core.storage_formats.pandas.parsers import (
+    ExperimentalCustomTextParser,
+    ExperimentalPandasPickleParser,
+    PandasCSVGlobParser,
```
Comment (YarShev) [inline modin/core/execution/dask/implementations/pandas_on_dask/io/io.py:52]:
```suggestion
    ExperimentalPandasCSVGlobParser,
```
Should we rename it? Separate issue or in this pr, I am okay with either option.

**Review by YarShev — APPROVED** [review]
LGTM! Left just one comment.

**Review by YarShev — APPROVED** [review]

**Comment by YarShev** [comment]
Btw, we have functionality in experimental.core folder that doesn't have Experimental prefix. We should align the naming there - either use the prefix for all objects or not.

**Comment by anmyachev** [comment]
@YarShev ready to merge

**Pull request merged**